src/settings.rs

```diff
@@ -105,9 +108,48 @@ impl kubewarden::settings::Validatable for Settings {
             return Err("Signatures must not be empty".to_string());
         }
 
-        // TODO: when a certificate is being used, ensure the certificate
-        // can be trusted. This requires a new waPC function being written
-
+        // when a certificate is being used, ensure it can be trusted
+        'signatures: for signature in self.signatures.iter() {
+            if let Signature::Certificate(s) = signature {
+                // build sdk structs:
+                let cert = SDKCert {
+                    encoding: CertificateEncoding::Pem,
+                    data: s.certificate.to_owned().into_bytes(),
+                };
+                let cert_chain_opt: Option<Vec<SDKCert>> = s.certificate_chain.as_ref().map({
+                    |chain| {
+                        chain
+                            .iter()
+                            .map(|c| SDKCert {
+                                encoding: CertificateEncoding::Pem,
+                                data: c.to_owned().into_bytes(),
+                            })
+                            .collect()
+                    }
+                });
+
+                match verify_cert(cert, cert_chain_opt, None) {
+                    Ok(b) => {
+                        match b {
+                            BoolWithReason::True => continue 'signatures, // cert verified
```

Comment:
Can't we rewrite the code like that:

```rust
BoolWithReason::True => return Ok(())
```

This would look more "linear" to me

Reply:
wouldn't that mean that we exit from the whole `validate()`, potentially leaving other signatures performed with certificates unchecked?
(if not, I wasn't sure, so I just did the label and the continue for clarity).

Reply:
yes, you're right. I take it back.